<a href="https://colab.research.google.com/github/ayelefeleke/Titanic-Data-Analysis/blob/main/Titanic_data_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [ ]:
data={
   "Student_Name": ["Ayele","Samuel","Segni","Nardos","Ifa","Bereket","Robera","Bonsa"
   ,"Ashanefi","Aron","Tolosa","Kidist","Selam","Fenet","Jemal"],

   "Department":["Software","CSE","CSE","Communication","power","Water","CSE","Mechanical",
                 "Material","Architecture","Software","Civil","Applied.Maths","Pharmacy","Pre_Engineering"],

   "Year":["Second year","Fifth year","Fifth year","Third year","Fourth year","Second year","Second year","Fourth year",
           "Fifth year","Third year","Fourth year","Second year","Second year","Third year","Freshman"],


   "Club in ASTU": ["Datascience","Datascience","Datascience","Student police","ASTU STEM","Student cafe","Datascience","Robotics",
                    "Datascience","Art","Cybersecurity","Development","Datascience","NO in Club","NO in club"],

   "CGPA":["3.92","3.75","3.68","3.43","3.00","2.90","2.89","4.00","3.91","3.78","3.68","3.84","3.97","3.65","3.96"]

}
df = pd.DataFrame(data)
df.index = [f"ID_{i+1}" for i in range(len(df))]
display(df)

,Student_Name,Department,Year,Club in ASTU,CGPA
ID_1,Ayele,Software,Second year,Datascience,3.92
ID_2,Samuel,CSE,Fifth year,Datascience,3.75
ID_3,Segni,CSE,Fifth year,Datascience,3.68
ID_4,Nardos,Communication,Third year,Student police,3.43
ID_5,Ifa,power,Fourth year,ASTU STEM,3.00
ID_6,Bereket,Water,Second year,Student cafe,2.90
ID_7,Robera,CSE,Second year,Datascience,2.89
ID_8,Bonsa,Mechanical,Fourth year,Robotics,4.00
ID_9,Ashanefi,Material,Fifth year,Datascience,3.91
ID_10,Aron,Architecture,Third year,Art,3.78


let first upload the file to see if the files are there

In [4]:
titanic = pd.read_csv("/content/train.csv")
titanic.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [ ]:
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [ ]:
titanic.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


this is the data cleaning

In [ ]:
# To Fill missing values
titanic['Age'] = titanic['Age'].fillna(titanic['Age'].median())
titanic['Embarked'] = titanic['Embarked'].fillna(titanic['Embarked'].mode()[0])

# To Drop Cabin column only if it exists
if 'Cabin' in titanic.columns:
    titanic = titanic.drop(columns=['Cabin'])

#To Remove duplicates
titanic = titanic.drop_duplicates()


let check if it is cleaned

cabin is successfulyy removed and the dataset is cleaned

let analysed the data

In [ ]:
survival_by_gender = titanic.groupby('Sex')['Survived'].mean()
print("Survival rate by gender:\n", survival_by_gender)

Survival rate by gender:
 Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64


In [ ]:
survival_by_class = titanic.groupby('Pclass')['Survived'].mean()
print("\nSurvival rate by class:\n", survival_by_class)


Survival rate by class:
 Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


In [ ]:
by_age_class = titanic.groupby('Pclass')['Age'].mean()
print("\nAverage age per class:\n", by_age_class)


Average age per class:
 Pclass
1    36.812130
2    29.765380
3    25.932627
Name: Age, dtype: float64


In [ ]:
# let Create age groups
bins = [0, 12, 18, 35, 60, 100]
labels = ['Child', 'Teen', 'Adult', 'Middle-Age', 'Senior']
titanic['AgeGroup'] = pd.cut(titanic['Age'], bins=bins, labels=labels)

survival_by_agegroup = titanic.groupby('AgeGroup')['Survived'].mean()
print("\nSurvival rate by age group:\n", survival_by_agegroup)


Survival rate by age group:
 AgeGroup
Child         0.579710
Teen          0.428571
Adult         0.353271
Middle-Age    0.400000
Senior        0.227273
Name: Survived, dtype: float64


/tmp/ipykernel_2567/1569839012.py:6: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  survival_by_agegroup = titanic.groupby('AgeGroup')['Survived'].mean()


filtering

In [ ]:
female_survived = titanic[(titanic['Sex'] == 'female') & (titanic['Survived'] == 1)]
display(female_survived)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Embarked,AgeGroup
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C,Middle-Age
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,S,Adult
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,S,Adult
8,9,1,3,"Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)",female,27.0,0,2,347742,11.1333,S,Adult
9,10,1,2,"Nasser, Mrs. Nicholas (Adele Achem)",female,14.0,1,0,237736,30.0708,C,Teen
...,...,...,...,...,...,...,...,...,...,...,...,...
874,875,1,2,"Abelson, Mrs. Samuel (Hannah Wizosky)",female,28.0,1,0,P/PP 3381,24.0000,C,Adult
875,876,1,3,"Najib, Miss. Adele Kiamie ""Jane""",female,15.0,0,0,2667,7.2250,C,Teen
879,880,1,1,"Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)",female,56.0,0,1,11767,83.1583,C,Middle-Age
880,881,1,2,"Shelley, Mrs. William (Imanita Parrish Hall)",female,25.0,0,1,230433,26.0000,S,Adult


In [ ]:
children_survived = titanic[(titanic['Age'] <= 12) & (titanic['Survived'] == 1)]
display(children_survived)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Embarked,AgeGroup
10,11,1,3,"Sandstrom, Miss. Marguerite Rut",female,4.00,1,1,PP 9549,16.7000,S,Child
43,44,1,2,"Laroche, Miss. Simonne Marie Anne Andree",female,3.00,1,2,SC/Paris 2123,41.5792,C,Child
58,59,1,2,"West, Miss. Constance Mirium",female,5.00,1,2,C.A. 34651,27.7500,S,Child
78,79,1,2,"Caldwell, Master. Alden Gates",male,0.83,0,2,248738,29.0000,S,Child
125,126,1,3,"Nicola-Yarred, Master. Elias",male,12.00,1,0,2651,11.2417,C,Child
165,166,1,3,"Goldsmith, Master. Frank John William ""Frankie""",male,9.00,0,2,363291,20.5250,S,Child
172,173,1,3,"Johnson, Miss. Eleanor Ileen",female,1.00,1,1,347742,11.1333,S,Child
183,184,1,2,"Becker, Master. Richard F",male,1.00,2,1,230136,39.0000,S,Child
184,185,1,3,"Kink-Heilmann, Miss. Luise Gretchen",female,4.00,0,2,315153,22.0250,S,Child
193,194,1,2,"Navratil, Master. Michel M",male,3.00,1,1,230080,26.0000,S,Child


In [ ]:
first_class_survived = titanic[(titanic['Pclass'] == 1) & (titanic['Survived'] == 1)]
display(first_class_survived)

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Embarked,AgeGroup
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C,Middle-Age
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,S,Adult
11,12,1,1,"Bonnell, Miss. Elizabeth",female,58.0,0,0,113783,26.5500,S,Middle-Age
23,24,1,1,"Sloper, Mr. William Thompson",male,28.0,0,0,113788,35.5000,S,Adult
31,32,1,1,"Spencer, Mrs. William Augustus (Marie Eugenie)",female,28.0,1,0,PC 17569,146.5208,C,Adult
...,...,...,...,...,...,...,...,...,...,...,...,...
862,863,1,1,"Swift, Mrs. Frederick Joel (Margaret Welles Ba...",female,48.0,0,0,17466,25.9292,S,Middle-Age
871,872,1,1,"Beckwith, Mrs. Richard Leonard (Sallie Monypeny)",female,47.0,1,1,11751,52.5542,S,Middle-Age
879,880,1,1,"Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)",female,56.0,0,1,11767,83.1583,C,Middle-Age
887,888,1,1,"Graham, Miss. Margaret Edith",female,19.0,0,0,112053,30.0000,S,Adult


now from the data set i can saw that
let answer the questions
1.who was more likely to surive ?
Ans.Females were more likely to survive than males
because as we heve seen from the above analysis
Survival rate for females is =0.742038,which is approximately 74%
whereas for males=0.188908,approximately 19%

2.Did the class affect survival ?
Yes->Passengers in higher classes(first class)had better  survival chances
 similarly as we seen
 1.first class survival is =0.62963,which is approximately 63%
 2.second class=0.472826,which is 47%
 3.3rd class =0.242363,which is approcimately 24%

 3.Where children prioritized ?
 Yes->children had a higher survival rate compared to adults
 e.g survival rate for children below 12 age is approximately 58%,while the rest have below this.

 4.What combination had the highest survival rate ?
 As we seen from the above questions the combinatioin of FEMALE,FIRST CLASS AND CHILD has the highest survival probability because each has highest survival rate in each of Gender,class,age respectively.
